codespell_lib/tests/test_basic.py

```diff
@@ -99,6 +99,10 @@ def test_basic(tmpdir, capsys):
     assert 'cannot find dictionary' in stderr
     os.remove(fname)
 
+    code, _, stderr = cs.main('--skip', 'file with [weird signs b-a].txt',
+                              f.name, std=True)
+    assert code == EX_USAGE, 'invalid glob'
```

Comment:
It would be nice to have the counter test that it passes when the escaping has been added!

Reply:
Just this please @DimitriPapadopoulos .

Reply:
I'm afraid I don't understand the request here. Which counter test are you referring to? Perhaps `count=True` in `MainWrapper.main()`?
https://github.com/codespell-project/codespell/blob/bb7d185a71acbe651608bf3527c1ac10cebe4ddf/codespell_lib/tests/test_basic.py#L25-L28

Reply:
Sorry, I mean anti/counter, i.e. can you successfully skip a file by escaping that character class's brackets.

Reply:
Mmmh.... I doubt casual users are able to escape the brackets. I had to read [`glob(7)`](https://man7.org/linux/man-pages/man7/glob.7.html) attentively:
> #### Character classes
>  An expression "`[...]`" where the first character after the leading '`[`' is not an '`!`' matches a single character, namely any of the characters enclosed by the brackets.  The string enclosed by the brackets cannot be empty; therefore '`]`' can be allowed between the brackets, provided that it is the first character.  (Thus, "`[][!]`" matches the three characters '`[`', '`]`', and '_!_'.)

to understand how to escape the brackets:
```python
>>> import fnmatch
>>> 
>>> filename = '[b-a].txt'
>>> 
>>> fnmatch.fnmatch(filename, '[[]b-a[]].txt')
True
>>> 
>>> fnmatch.fnmatch(filename, '[b-a].txt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/fnmatch.py", line 36, in fnmatch
    return fnmatchcase(name, pat)
  File "/usr/lib/python3.8/fnmatch.py", line 70, in fnmatchcase
    match = _compile_pattern(pat)
  File "/usr/lib/python3.8/fnmatch.py", line 46, in _compile_pattern
    return re.compile(res).match
  File "/usr/lib/python3.8/re.py", line 252, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python3.8/re.py", line 304, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.8/sre_compile.py", line 764, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.8/sre_parse.py", line 948, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/lib/python3.8/sre_parse.py", line 443, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
  File "/usr/lib/python3.8/sre_parse.py", line 834, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
  File "/usr/lib/python3.8/sre_parse.py", line 443, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
  File "/usr/lib/python3.8/sre_parse.py", line 598, in _parse
    raise source.error(msg, len(this) + 1 + len(that))
re.error: bad character range b-a at position 5
>>> 
```

Reply:
Test with escaped brackets works now.

Reply:
@peternewman I have applied the requested change.